pattern/constraint/thing/ValueConstraint.java

```diff
@@ -195,6 +198,16 @@ public FunctionalIterator<AlphaEquivalence> alphaEquals(ValueConstraint<?> that)
                 .flatMap(a -> a.alphaEqualIf(this.value.equals(that.value)));
     }
 
+    public boolean maybeCompatible(ValueConstraint<?> valueConstraint) {
+        if (valueConstraint.predicate == EQ) {
+            return isCompatibleWithEQ(valueConstraint);
+        } else { // TODO: implement inequality compatibility when useful
+            return true;
```

Comment:
Implementing this may not be hard for the (in)Equality constraints , but may be hopelessly complicated and uninformative in the case of SubString constraints. 
There is also no use case for it at the moment, so we defer implementing it till we need it.

Reply:
useful comment, good!